cookiecutter/prompt.py

```diff
@@ -313,3 +341,36 @@ def prompt_for_config(context, no_input=False):
             raise UndefinedVariableInTemplate(msg, err, context) from err
 
     return cookiecutter_dict
+
+
+def choose_nested_template(context: dict, repo_dir: str, no_input: bool = False) -> str:
+    """Prompt user to select the nested template to use.
+
+    :param context: Source for field names and sample values.
+    :param repo_dir: Repository directory.
+    :param no_input: Do not prompt for user input and use only values from context.
+    :returns: Path to the selected template.
+    """
+    cookiecutter_dict = OrderedDict([])
+    env = StrictEnvironment(context=context)
+    prefix = ""
+    prompts = _extract_prompts(context)
+    key = "templates"
+    config = context['cookiecutter'].get(key, {})
+    if config:
+        # Pass
+        val = prompt_choice_for_template(key, config, no_input)
+        template = config[val]["path"]
+    else:
+        # Old style
+        key = "template"
+        config = context['cookiecutter'].get(key)
+        val = prompt_choice_for_config(
+            cookiecutter_dict, env, key, config, no_input, prompts, prefix
+        )
+        template = re.search(r'\((.*?)\)', val).group(1)
```

Comment:
This will match on `"()"` and result in `template_path` being the same value as `repo_dir`. Is this the desired behavior?

It's also possible that a user could specify `"(/)"`, which `repo_dir / template` will dutifully resolve to the root directory. You can check this by doing something like:

```python
template_path = pathlib.Path(template)
if template_path.is_absolute():
    raise ValueError('hang on there, buddy')  # or make it relative
```

It's also possible that this won't match on anything, `re.search()` will return `None`, and `.group(1)` will throw an `AttributeError`.

I recommend throwing strange values at this and confirming that it's not possible to generate paths that might lead to e.g. malicious changes to the filesystem.